### Part 1 — Prepare Your Data

In [1]:
import pandas as pd
df=pd.read_csv("Sales_Data_Cleaned")

In [2]:
df

,Order_ID,Date,Product,Category,Quantity,Price,City,OrderYear,OrderMonth,OrderDay
0,1001,2026-01-11,Mobile,Electronics,3.0,42750.0,Lahore,2026,1,11
1,1002,2026-01-31,Headphones,Electronics,5.0,6175.0,Peshawar,2026,1,31
2,1003,2026-01-13,Printer,Office,1.0,26600.0,Lahore,2026,1,13
3,1004,2026-05-14,Keyboard,Accessories,5.0,4275.0,Peshawar,2026,5,14
4,1005,2026-06-20,Keyboard,Accessories,5.0,4500.0,Lahore,2026,6,20
5,1006,2026-06-04,Notebook,Stationery,3.0,475.0,Lahore,2026,6,4
6,1007,2026-04-02,Printer,Office,3.0,26600.0,Lahore,2026,4,2
7,1008,2026-01-31,Monitor,Electronics,3.0,32000.0,Islamabad,2026,1,31
8,1009,2026-04-03,Monitor,Electronics,5.0,32000.0,Islamabad,2026,4,3
9,1010,2026-05-22,Notebook,Stationery,1.0,500.0,Islamabad,2026,5,22


In [3]:
df.dtypes

Order_ID        int64
Date           object
Product        object
Category       object
Quantity      float64
Price         float64
City           object
OrderYear       int64
OrderMonth      int64
OrderDay        int64
dtype: object

In [4]:
df["Revenue"]=df["Quantity"]*df["Price"]

In [5]:
df["Date"]=pd.to_datetime(df["Date"])

In [6]:
df["MonthName"]=df["Date"].dt.month_name()

### Part 2 — Revenue Analysis

Q1

Calculate total revenue for each product.

Then identify:

Highest-revenue product
Lowest-revenue product

In [7]:
df.groupby("Product")["Revenue"].sum()

Product
Headphones      89700.0
Keyboard       163575.0
Laptop        1011500.0
Mobile        1759500.0
Monitor        612800.0
Mouse           29625.0
Notebook        17925.0
Printer        495600.0
Name: Revenue, dtype: float64

Mobile generates the highest revenue, followed by Laptop. Together, these products account for a substantial share of total revenue.

Q2

Calculate the percentage contribution of each product to total revenue

In [8]:
product_revenue=df.groupby("Product")["Revenue"].sum()

In [9]:
percentage=product_revenue/df["Revenue"].sum() * 100

In [10]:
print(percentage)

Product
Headphones     2.145818
Keyboard       3.913067
Laptop        24.197262
Mobile        42.091036
Monitor       14.659498
Mouse          0.708694
Notebook       0.428805
Printer       11.855821
Name: Revenue, dtype: float64


Mobiles are generating 42% of the total revenue followed by Laptops(24.20%) and Monitor(14.66%).
Accessories and Stationary products are lowest in terms of percentage contribution in the total revenue. 

Q3

Calculate total revenue by city.

Then identify:

Highest revenue city

Lowest revenue city

In [11]:
df.groupby("City")["Revenue"].sum()

City
Islamabad     1393625.0
Karachi        587800.0
Lahore         590875.0
Multan         730325.0
Peshawar       387075.0
Rawalpindi     490525.0
Name: Revenue, dtype: float64

Highest revenue city: Islamabad(1393625.0)

Lowest revenue city: Peshawar(387075.0)

Q4

Calculate total revenue by category.

Then calculate each category's percentage contribution to total revenue.

In [12]:
Category_Revenue=df.groupby("Category")["Revenue"].sum()

In [13]:
per_bycategory=Category_Revenue/df["Revenue"].sum() * 100

In [14]:
print(per_bycategory)

Category
Accessories     4.621761
Electronics    83.093613
Office         11.855821
Stationery      0.428805
Name: Revenue, dtype: float64


Electronics are contributing the highest share (83.09%) in the total revenue followed by office equipments that are contributing 11.86%.

Electronics dominate the company's revenue, suggesting that overall sales performance is heavily dependent on this category.

### Part 3 — Quantity vs Revenue

Q5

Create a table showing each product's:

Total Quantity,
Total Revenue,
Average Price

In [15]:
product_summary = df.groupby("Product").agg(
    Total_Quantity=("Quantity", "sum"),
    Total_Revenue=("Revenue", "sum"),
    Average_Price=("Price", "mean")
)

product_summary

,Total_Quantity,Total_Revenue,Average_Price
Product,,,
Headphones,14.0,89700.0,6391.666667
Keyboard,37.0,163575.0,4413.461538
Laptop,12.0,1011500.0,83300.000000
Mobile,39.0,1759500.0,45187.500000
Monitor,19.0,612800.0,32320.000000
Mouse,12.0,29625.0,2500.000000
Notebook,37.0,17925.0,488.636364
Printer,18.0,495600.0,27533.333333


Q6

Find out:

Does the product with the highest quantity also generate the highest revenue?

Write a short explanation.

No. The product with the highest quantity does not necessarily generate the highest revenue. Revenue depends on both quantity sold and price per unit.

Q7 — Challenge ⭐

Find products that have:

High quantity but relatively low revenue.

Explain why this might happen.

Think about:

Price
Quantity
Product type

In [16]:
product_summary.sort_values(
    by="Total_Quantity",
    ascending=False
)

,Total_Quantity,Total_Revenue,Average_Price
Product,,,
Mobile,39.0,1759500.0,45187.500000
Keyboard,37.0,163575.0,4413.461538
Notebook,37.0,17925.0,488.636364
Monitor,19.0,612800.0,32320.000000
Printer,18.0,495600.0,27533.333333
Headphones,14.0,89700.0,6391.666667
Laptop,12.0,1011500.0,83300.000000
Mouse,12.0,29625.0,2500.000000


Notebook has relatively high sales volume, but its revenue contribution is very small because its selling price is much lower than products such as Mobile, Laptop and Monitor.

### Part 4 — Monthly Analysis 📅

Q8

Calculate:

Monthly total revenue

and sort it from highest to lowest.

In [17]:
df.groupby("MonthName")["Revenue"].sum().sort_values(ascending=False)

MonthName
April       978225.0
June        880450.0
January     877900.0
March       649650.0
February    491400.0
May         302600.0
Name: Revenue, dtype: float64

Highest revenue generated in April(978225.0) followed by June(880450.0) and January(880450.0)

9

Find the month with:

Highest revenue
Lowest revenue

Highest revenue: April(978225.0)

Lowest revenue:May(302600.0)

Q10

Calculate monthly:

Average order revenue

Then compare it with monthly total revenue.

In [18]:
df.groupby("MonthName")["Revenue"].mean()

MonthName
April       122278.125000
February     54600.000000
January      79809.090909
June         88045.000000
March        64965.000000
May          25216.666667
Name: Revenue, dtype: float64

In [19]:
Revenue_Summary=df.groupby("MonthName").agg(
    Average_Revenue=("Revenue","mean"),
    Total_Revenue=("Revenue","sum")
)

Revenue_Summary
    

,Average_Revenue,Total_Revenue
MonthName,,
April,122278.125000,978225.0
February,54600.000000,491400.0
January,79809.090909,877900.0
June,88045.000000,880450.0
March,64965.000000,649650.0
May,25216.666667,302600.0


Does the month with the highest total revenue also have the highest average order revenue?

yes, highest total revenue also have the highest average order revenue.

### Part 5 — City Analysis

Q11

Create a table showing each city's:

Total Revenue,
Number of Orders,
Average Order Revenue,
Total Quantity.

In [20]:
City_Revenue=df.groupby("City").agg(
    Total_Revenue=("Revenue","sum"),
    No_of_Orders=("Order_ID","count"),
    Average_Revenue=("Revenue","mean"),
    Total_Quantity=("Quantity","sum")
)

City_Revenue

,Total_Revenue,No_of_Orders,Average_Revenue,Total_Quantity
City,,,,
Islamabad,1393625.0,12,116135.416667,41.0
Karachi,587800.0,8,73475.000000,28.0
Lahore,590875.0,13,45451.923077,42.0
Multan,730325.0,12,60860.416667,33.0
Peshawar,387075.0,9,43008.333333,27.0
Rawalpindi,490525.0,6,81754.166667,17.0


Q12

Find:

Which city has the highest number of orders?

Then ask:

Is it also the city with the highest revenue?

Explain your answer.

Lahore has the highest number of orders, but its revenue is lower than Islamabad and Multan. This suggests that the average value of orders and product mix differ across cities.

Q13 — Business Thinking ⭐

Imagine your manager says:

"We want to expand into one city. Which city would you recommend?"

Choose one city and support your recommendation with at least two metrics.

Don't simply choose the city with the highest revenue.

In [21]:
df.groupby("Product")["Revenue"].sum()

Product
Headphones      89700.0
Keyboard       163575.0
Laptop        1011500.0
Mobile        1759500.0
Monitor        612800.0
Mouse           29625.0
Notebook        17925.0
Printer        495600.0
Name: Revenue, dtype: float64

In [22]:
df.groupby(["City","Product"])["Quantity"].sum()

City        Product   
Islamabad   Headphones     4.0
            Keyboard       3.0
            Laptop         7.0
            Mobile         8.0
            Monitor       13.0
            Notebook       6.0
Karachi     Headphones     5.0
            Keyboard       4.0
            Laptop         4.0
            Mobile         4.0
            Mouse          3.0
            Notebook       8.0
Lahore      Keyboard       9.0
            Mobile         3.0
            Monitor        3.0
            Notebook      15.0
            Printer       12.0
Multan      Keyboard       8.0
            Mobile         9.0
            Monitor        3.0
            Mouse          2.0
            Notebook       5.0
            Printer        6.0
Peshawar    Headphones     5.0
            Keyboard       8.0
            Laptop         1.0
            Mobile         5.0
            Mouse          5.0
            Notebook       3.0
Rawalpindi  Keyboard       5.0
            Mobile        10.0
            Mous

I recommend Islamabad because it has the highest total revenue and highest average order revenue. This suggests both strong market demand and higher-value orders.

### Part 6 — Product × City Analysis

Q14

Create a Pivot Table:

Rows =City,
Columns = Product,
Values = Revenue,
Aggregation  =Sum.

In [23]:
df.pivot_table(values="Revenue",index="City",columns="Product", aggfunc="sum")

Product,Headphones,Keyboard,Laptop,Mobile,Monitor,Mouse,Notebook,Printer
City,,,,,,,,
Islamabad,24700.0,13050.0,595000.0,342000.0,416000.0,NaN,2875.0,NaN
Karachi,34125.0,18000.0,335750.0,189000.0,NaN,7125.0,3800.0,NaN
Lahore,NaN,40050.0,NaN,128250.0,96000.0,NaN,7375.0,319200.0
Multan,NaN,36000.0,NaN,409500.0,100800.0,5250.0,2375.0,176400.0
Peshawar,30875.0,34200.0,80750.0,227250.0,NaN,12500.0,1500.0,NaN
Rawalpindi,NaN,22275.0,NaN,463500.0,NaN,4750.0,NaN,NaN


Q15

Using this table, answer:

Which city appears strongest for Mobile sales?

Rawalpindi has more demand and sales  for mobile phones.Total Revenue from this city is aproximately 463500.0.

Q16

Find the highest-revenue City + Product combination.

For example:

Islamabad + Laptop

Multan is the second one in terms of contribution to total revenue with higher demand for Mobile + Monitor. Lahore is on the third no with higher demand for printers.


### Part 7 — Your Own EDA

1.Which month has higher demand for products?

2.Does this month has also higher total revenue?

3. Demand for different products in all months?

In [24]:
df.groupby("MonthName")["Quantity"].sum()

MonthName
April       28.0
February    33.0
January     36.0
June        35.0
March       24.0
May         32.0
Name: Quantity, dtype: float64

January shows the higher demand for products followed by June and feb.

In [25]:
Revenue_Summary=df.groupby("MonthName").agg(
    Average_Revenue=("Revenue","mean"),
    Total_Revenue=("Revenue","sum")
)

Revenue_Summary
    

,Average_Revenue,Total_Revenue
MonthName,,
April,122278.125000,978225.0
February,54600.000000,491400.0
January,79809.090909,877900.0
June,88045.000000,880450.0
March,64965.000000,649650.0
May,25216.666667,302600.0


This data shows that higher demand for products in a specific month does not relate to the monthly revenue as April has the highest total revenue with
min total quantity sold(28).

In [26]:
df.pivot_table(values="Quantity",index="MonthName",columns="Product", aggfunc="sum")

Product,Headphones,Keyboard,Laptop,Mobile,Monitor,Mouse,Notebook,Printer
MonthName,,,,,,,,
April,NaN,1.0,3.0,7.0,10.0,NaN,4.0,3.0
February,NaN,11.0,NaN,2.0,6.0,NaN,9.0,5.0
January,5.0,1.0,NaN,16.0,3.0,2.0,8.0,1.0
June,NaN,10.0,3.0,10.0,NaN,5.0,3.0,4.0
March,5.0,5.0,6.0,2.0,NaN,1.0,5.0,NaN
May,4.0,9.0,NaN,2.0,NaN,4.0,8.0,5.0


There is relatively more demand for electronic products in April,June and Jan that lead to the higher revenue generation in these months.

### Part 8 — Management Summary

Key Findings

1. Total sales of the company do not show any specific trend except the months with higher demand for electronics generated the higher revenues.
    
2.Electronic products like monitor, laptop and mobiles are contributing the highest share (83.09%) in the total revenue followed by office equipments like printer that are contributing 11.86%.

3.Highest revenue city is Islamabad(1393625.0) and Lowest revenue city is the Peshawar(387075.0).

4.Highest revenue generated in April(978225.0) followed by June(880450.0) and January(880450.0).May shows the min total revenue.

5.Islamabad the largest market shows the higher demand for Laptop.Rawalpindi has more demand and sales for mobile phones.Total Revenue from this city is aproximately 463500.0.Multan is the second one in terms of contribution to total revenue with higher demand for Mobile + Monitor. Lahore is on the third no with higher demand for printers.

Recommendations:

As 

1.Sales of electronic products are contributing more in the total revenue (Laptop 1011500.0, Mobile 1759500.0, Monitor 612800.0 )that is why company should focus on these products.

2.There is also higher demand for these products in Multan and Rawalpindi.More Revenue can be generate by increase in demand in these cities through different promotional stretegies.





### EDA Day 2 Challenge

In [27]:

product_summary = df.groupby("Product").agg(
    Total_Quantity=("Quantity", "sum"),
    Total_Revenue=("Revenue", "sum"),
    Average_Price=("Price", "mean")
)

product_summary["Revenue_per_unit"] = (
    product_summary["Total_Revenue"] /
    product_summary["Total_Quantity"]
)

product_summary

,Total_Quantity,Total_Revenue,Average_Price,Revenue_per_unit
Product,,,,
Headphones,14.0,89700.0,6391.666667,6407.142857
Keyboard,37.0,163575.0,4413.461538,4420.945946
Laptop,12.0,1011500.0,83300.000000,84291.666667
Mobile,39.0,1759500.0,45187.500000,45115.384615
Monitor,19.0,612800.0,32320.000000,32252.631579
Mouse,12.0,29625.0,2500.000000,2468.750000
Notebook,37.0,17925.0,488.636364,484.459459
Printer,18.0,495600.0,27533.333333,27533.333333


Laptop has higher revenue per unit followed by mobiles and monitors. The months(April,June,Jan) with higher demand for these products generated the highest revenues.